### PR TITLE
LIME-249: Add Public Profile page

### LINKED ISSUE #249: Add Public profile page for Social feature
**SPECS from QA

Social features - Public Profile Page :**

**1. Personal Details Section:**

- Profile Picture
- Full Name with Nickname below if the last exists 
- Location (will be added to profile page)
- Preferences: if they are not defined, may be retrieved from goals

**2. Social media buttons section**

- Add friend
- Message

**3. Last Workout data**

- Name of activity
- Distance
- Duration
- Time when it was finished

**4. Statistics Section for current month:**

- Total number of workouts
- Total duration of workouts
- Total distance (in km or steps) 
- Total calories concumed

**5. Achievements section**
List of all achievements of the user (like on the Goals page)

Image is just a sample of possible location of sections

Image

### PR DESCRIPTION
**Pull request by valery-chumak**

### CODE REVIEW

**Comment by valery-chumak** [comment]
For now Location is just a mocked data
image
image
image
image
image

**Comment by valery-chumak** [comment]
Light theme
image
image

**Comment by nikita-remeslov** [comment]
Discussed in slack

**Comment by valery-chumak** [comment]
Add new widget color
image

File: frontend/src/bundles/common/helpers/calculate-age/calculate-age.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const calculateAge = (dateOfBirth: string | null): number | null => {
```
Comment (dimapopovych) [inline frontend/src/bundles/common/helpers/calculate-age/calculate-age.ts]: file name should have helpers suffix
Also I think could be simplified by using date-fns

File: frontend/src/bundles/common/helpers/get-last-workout/get-last-workout.helper.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { type WorkoutResponseDto } from '~/bundles/workouts/types/types.js';
+
+const getLastWorkout = (
+    workouts: WorkoutResponseDto[],
+): WorkoutResponseDto | null => {
+    if (!workouts || workouts.length === 0) {
+        return null;
+    }
+    const sortedWorkouts = [...workouts].sort((a, b) => {
```
Comment (dimapopovych) [inline frontend/src/bundles/common/helpers/get-last-workout/get-last-workout.helper.ts]: probably we could already use toSorted

File: frontend/src/bundles/common/helpers/get-last-workout/get-last-workout.helper.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        return (
+            new Date(b.workoutStartedAt).getTime() -
+            new Date(a.workoutStartedAt).getTime()
+        );
+    });
+
+    const lastWorkout = sortedWorkouts[0];
+    if (!lastWorkout) {
+        return null;
+    }
```
Comment (dimapopovych) [inline frontend/src/bundles/common/helpers/get-last-workout/get-last-workout.helper.ts]: why do we need this check? We already asserted that there is !== 0 workouts.
I would suggest to change a little
0 workouts - return null
1 workout - return this workout immediately
in other case sort

File: frontend/src/bundles/profile/pages/public-profile-page/public-profile-page.tsx [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    );
+    const { dataStatus: dataStatusAchievements, achievements } = useAppSelector(
+        ({ achievements }) => achievements,
+    );
+    const { dataStatus: dataStatusWorkouts, workouts } = useAppSelector(
+        ({ workouts }) => workouts,
+    );
+    const { dataStatus: dataStatusGoals, goals } = useAppSelector(
+        ({ goals }) => goals,
+    );
```
Comment (dimapopovych) [inline frontend/src/bundles/profile/pages/public-profile-page/public-profile-page.tsx:62]: Maybe it's worth to create separate slice for this page?

**Review by dimapopovych — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/profile/pages/public-profile-page/public-profile-page.tsx (hunk 4 above)
Comment (valery-chumak) [reply frontend/src/bundles/profile/pages/public-profile-page/public-profile-page.tsx:62]: As I know for other pages we didn't create a separate slices, wouldn't it be confusing if I create it here?

**Review by nikita-remeslov — DISMISSED** [review]

File: frontend/src/bundles/profile/pages/public-profile-page/public-profile-page.tsx [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+    const isLoading =
+        dataStatusUser === DataStatus.PENDING ||
+        dataStatusAchievements === DataStatus.PENDING ||
+        dataStatusWorkouts === DataStatus.PENDING ||
+        dataStatusGoals === DataStatus.PENDING;
```
Comment (jhladkov) [inline frontend/src/bundles/profile/pages/public-profile-page/public-profile-page.tsx]: I think we can simplify it like this:
 `[dataStatusUser, dataStatusAchievements, dataStatusWorkouts, dataStatusGoals].includes(DataStatus.PENDING);`
 But double-check it )

File: frontend/src/bundles/profile/pages/public-profile-page/public-profile-page.tsx [hunk 6]
```diff
@@ -0,0 +1,185 @@
+import { actions as achievementsActions } from '~/bundles/achievements/store/achievements.js';
+import {
+    AchievementCard,
+    ActivityWidget,
+    ActivityWidgetColor,
+    Icon,
+    Loader,
+} from '~/bundles/common/components/components.js';
+import { IconColor } from '~/bundles/common/components/icon/enums/icon-colors.enum.js';
+import { IconName } from '~/bundles/common/components/icon/enums/icon-name.enum.js';
```
Comment (jhladkov) [inline frontend/src/bundles/profile/pages/public-profile-page/public-profile-page.tsx]: Can we combine it?
 `import { IconColor, IconName  } from '~/bundles/common/components/icon/enums/enums.js';`

File: frontend/src/bundles/profile/pages/public-profile-page/public-profile-page.tsx (hunk 5 above)
Comment (valery-chumak) [reply frontend/src/bundles/profile/pages/public-profile-page/public-profile-page.tsx]: Thanks) Fixed. Working correct)

**Review by nikita-remeslov — DISMISSED** [review]

File: frontend/src/bundles/common/helpers/get-last-workout/get-last-workout.helper.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+): WorkoutResponseDto | null => {
+    if (workouts.length === 0) {
+        return null;
+    }
+
+    if (workouts.length === 1) {
+        return workouts[0] as WorkoutResponseDto;
+    }
+
+    const sortedWorkouts = [...workouts].toSorted((a, b) => {
```
Comment (dimapopovych) [inline frontend/src/bundles/common/helpers/get-last-workout/get-last-workout.helper.ts]: spread is redundant

**Review by dimapopovych — DISMISSED** [review]

**Pull request merged**